src/aiida/brokers/rabbitmq/defaults.py

```diff
@@ -29,23 +29,29 @@
 )
 
 
-def detect_rabbitmq_config(**kwargs) -> dict[str, t.Any] | None:
+def detect_rabbitmq_config(
+    protocol: str | None = None,
+    username: str | None = None,
+    password: str | None = None,
+    host: str | None = None,
+    port: int | None = None,
+    virtual_host: str | None = None,
+    heartbeat: int | None = None,
+) -> dict[str, t.Any] | None:
     """Try to connect to a RabbitMQ server with the default connection parameters.
 
     :returns: The connection parameters if the RabbitMQ server was successfully connected to, or ``None`` otherwise.
     """
     from kiwipy.rmq.threadcomms import connect
 
     connection_params = {
-        'protocol': kwargs.get('broker_protocol', os.getenv('AIIDA_BROKER_PROTOCOL', BROKER_DEFAULTS['protocol'])),
-        'username': kwargs.get('broker_username', os.getenv('AIIDA_BROKER_USERNAME', BROKER_DEFAULTS['username'])),
-        'password': kwargs.get('broker_password', os.getenv('AIIDA_BROKER_PASSWORD', BROKER_DEFAULTS['password'])),
-        'host': kwargs.get('broker_host', os.getenv('AIIDA_BROKER_HOST', BROKER_DEFAULTS['host'])),
-        'port': kwargs.get('broker_port', os.getenv('AIIDA_BROKER_PORT', BROKER_DEFAULTS['port'])),
-        'virtual_host': kwargs.get(
-            'broker_virtual_host', os.getenv('AIIDA_BROKER_VIRTUAL_HOST', BROKER_DEFAULTS['virtual_host'])
-        ),
-        'heartbeat': kwargs.get('broker_heartbeat', os.getenv('AIIDA_BROKER_HEARTBEAT', BROKER_DEFAULTS['heartbeat'])),
+        'protocol': protocol or os.getenv('AIIDA_BROKER_PROTOCOL', BROKER_DEFAULTS['protocol']),
+        'username': username or os.getenv('AIIDA_BROKER_USERNAME', BROKER_DEFAULTS['username']),
+        'password': password or os.getenv('AIIDA_BROKER_PASSWORD', BROKER_DEFAULTS['password']),
+        'host': host or os.getenv('AIIDA_BROKER_HOST', BROKER_DEFAULTS['host']),
+        'port': port or os.getenv('AIIDA_BROKER_PORT', BROKER_DEFAULTS['port']),
```

Comment:
I am not a 100% sure but I think for `port` and `heartbeat` we might need to cast to `int` when the value comes from `os.getenv` because I think those values are always strings. Not sure if down the line strings are accepted or anyway automatically cast.

Reply:
I think I tried passing strings accidentally at some point, and it didn't work. I'll recast them here to be sure.